modules/etc/image_etc.cpp

```diff
@@ -139,6 +139,13 @@ static void _compress_etc(Image *p_img, float p_lossy_quality, bool force_etc1_f
 		return;
 	}
 
+	if (img_format >= Image::FORMAT_RGBA8 && force_etc1_format) {
```

Comment:
i think you should check for alpha detected instead, check the detect_channels structure

Reply:
I initially used `` if (detected_channels == Image::DETECTED_RGBA && force_etc1_format)`` however, the branch was never run for some reason so I switched to using what I have here. I'll take a deeper look if it is truly better to use ``DETECTED_RGBA``.

Reply:
Fixed in https://github.com/godotengine/godot/pull/34790